airflow/providers/google/cloud/hooks/compute_engine_ssh.py

```diff
@@ -0,0 +1,211 @@
+
+        :param oslogin:
+        :param account:
+        :param private_key_file:
+        :param expire_time:
+        """
+        private_key_file = private_key_file or '/tmp/key-' + str(uuid.uuid4())
+        self.execute_in_compute(['ssh-keygen', '-t', 'rsa', '-N', '', '-f', private_key_file])
+
+        with open(private_key_file + '.pub', 'r') as original:
+            public_key = original.read().strip()
+
+        # Expiration time is in microseconds.
+        expiration = int((time.time() + expire_time) * 1000000)
+
+        body = {
+            'key': public_key,
+            'expirationTimeUsec': expiration,
+        }
+        oslogin.users().importSshPublicKey(parent=account, body=body).execute_in_compute()
+        return private_key_file
+
+    @GoogleBaseHook.fallback_to_default_project_id
+    def run(self,
+            cmd: str, project_id: str, instance: str, zone: str,
+            private_key_file: Optional[str] = None, expire_time: int = 300,
+            hostname: Optional[str] = None, tunnel: bool = False,
+            clean_up_private_key: bool = True):
+        """
+        Run a command on a remote system.
+
+        :param cmd:
+        :param project_id:
+        :param instance:
+        :param zone:
+        :param private_key_file:
+        :param expire_time:
+        :param hostname:
+        :param tunnel:
+        :param clean_up_private_key:
+        :return:
+        """
+
+        hostname = self.get_hostname(project_id=project_id,
+                                     instance=instance,
+                                     zone=zone,
+                                     hostname=zone)
+
+        account = self.get_service_account()
+        oslogin = self.get_oslogin()
+
+        private_key_file = self.create_ssh_key(oslogin=oslogin,
+                                               account=account,
+                                               private_key_file=private_key_file,
+                                               expire_time=expire_time)
+
+        username = self.get_username()
+
+        if tunnel and "--tunnel-through-iap" not in cmd:
+            cmd = cmd + " " + "--tunnel-through-iap"
+
+        ssh_command = [
+            'ssh', '-i', private_key_file, '-o', 'StrictHostKeyChecking=no',
+            f'{username}@{hostname}',
+            cmd,
+        ]
+        ssh: Any = subprocess.Popen(
+            ssh_command, shell=False, stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE)
+        result = ssh.stdout.readlines()
+        if clean_up_private_key:
+            self.clean_up(private_key_file)
+        return result if result else ssh.stderr.readlines()
+
+    def clean_up(self, private_key_file: str) -> None:
+        """
+        Removes created private key file
+
+        :param private_key_file:
+        :type: str
+        """
+        self.execute_in_compute(['shred', private_key_file])
+        self.execute_in_compute(['rm', private_key_file])
```

Comment:
If possible, we should avoid saving the private key to disk.

Reply:
It is actually deleted even when the user didn't instruct for it to be removed. Once the expire time has elapsed, the key is deleted even when the clean_up_private_key is False but I will check if it can work without saving to disk.